ui/src/message_popup/mobile_prototype/mobile_interface.jsx

```diff
@@ -0,0 +1,375 @@
+  },
+
+  setPassedResponse(){
+    this.setState({revisedResponse: ''});
+  },
+  
+  isReadyToMoveOn(){
+    if(this.props.scaffolding.helpType === 'feedback'){
+      if(this.state.revisedResponse !== undefined){ 
+        return true;
+      }
+    }else{
+      if(this.state.initialResponse !== undefined){
+        return true;
+      }
+    }
+    return false;
+  },
+  
+  onNextQuestion(){
+    this.props.onQuestionDone(this.state.elapsedMs);
+    this.setState(this.getInitialState());
+  },
+  
+  sendRevision(response){
+    if(this.props.question.examples.length > 0){
+      this.addMessages([
+        {type: 'user', text: response},
+        {type: 'info', text: "Here's an example of something you could say/do:\n\n" + _.shuffle(this.props.question.examples)[0] + "\n\nWould you like to revise your response?"}, 
+      ]);
+    }else{
+      this.addMessages([
+        {type: 'user', text: response},
+        {type: 'info', text: "There aren't any examples for this situation yet but your response will be recorded for feedback."}, 
+      ]);
+      this.setPassedResponse();
+    }
+  },
+  
+  showExample(){
+    const example = this.nextExample(true);
+    const messageText = example.type + ' Example:\n' + example.text;
+    this.addMessages([{type: 'info', text: messageText}]);
+  },
+  
+  nextExample(increment){
+    const currentIndex = this.state.exampleIndex;
+    if(this.state.allExamples.length > currentIndex){
+      if(increment){
+        this.setState({exampleIndex: this.state.exampleIndex + 1});
+      }
+      return this.state.allExamples[currentIndex];
+    }else{
+      return null;
+    }
+  },
+
+  setDialog(value){
+    this.setState({dialog: value});
+  },
+
+  onCloseDialog(){
+    this.setDialog('none');
+  },
+
+  onOpenInfoDialog(){
+    this.setDialog('info');
+  },
+
+  onOpenStudentDialog(student){
+    return (function() {
+      this.setState({selectedStudent: student});
+      this.setDialog('student');
+    }.bind(this));
+  },
+
+  logResponse() {
+    this.logData('message_popup_response');
+  },
+  
+  logRevision(finalText) {
+    this.logData('message_popup_revision', {finalResponseText: finalText});
+  },
+  
+  logRevisionDeclined(){
+    this.logData('message_popup_revision_declined');
+  },
+  
+  logData(type, params = {}) {
+    const {elapsedMs, initialResponse} = this.state;
+    const {question, scaffolding} = this.props;
+    const {shouldShowStudentCard, helpType} = scaffolding;
+    const {finalResponseText} = params;
+    const response:Response = {
+      question,
+      shouldShowStudentCard,
+      helpType,
+      elapsedMs,
+      initialResponseText: initialResponse,
+      finalResponseText
+    };
+    this.props.onLog(type, response);
+  },
+
+  getInitialMessages(questionObject, shouldShowStudentCard){
+    var messages = [];
+    if(_.has(questionObject, 'encodedText')){
+      var arrayOfMessages = questionObject.encodedText.split('[:');
```

Comment:
Encoding this structure in the questions directly will let you remove this parsing code altogether :)